src/main/java/org/opensearch/security/support/WildcardMatcher.java

```diff
@@ -150,7 +150,9 @@ public String toString() {
     };
 
     public static WildcardMatcher from(String pattern, boolean caseSensitive) {
-        if (pattern.equals("*")) {
+        if (pattern == null) {
```

Comment:
In the case of empty string, the wildcard matcher would be an ExactMatcher and the only pattern it would match is another empty string. IMO its already being handled as expected.

Reply:
okay, makes sense